Mathlib/AlgebraicTopology/SimplicialSet/HomotopyCat.lean

```diff
@@ -0,0 +1,363 @@
+
+/-- The middle vertex of a 2-simplex in a 2-truncated simplicial set. -/
+def ev1₂ {V : SSet.Truncated 2} (φ : V _[2]₂) : OneTruncation₂ V := V.map ι1₂.op φ
+
+/-- The final vertex of a 2-simplex in a 2-truncated simplicial set. -/
+def ev2₂ {V : SSet.Truncated 2} (φ : V _[2]₂) : OneTruncation₂ V := V.map ι2₂.op φ
+
+/-- The 0th face of a 2-simplex, as a morphism in the 2-truncated simplex category. -/
+def δ0₂ : [1]₂ ⟶ [2]₂ := δ₂ (n := 1) 0
+
+/-- The 1st face of a 2-simplex, as a morphism in the 2-truncated simplex category. -/
+def δ1₂ : [1]₂ ⟶ [2]₂ := δ₂ (n := 1) 1
+
+/-- The 2nd face of a 2-simplex, as a morphism in the 2-truncated simplex category. -/
+def δ2₂ : [1]₂ ⟶ [2]₂ := δ₂ (n := 1) 2
+
+/-- The arrow in the ReflQuiver `OneTruncation₂ V` of a 2-truncated simplicial set arising from the
+0th face of a 2-simplex. -/
+def ev12₂ {V : SSet.Truncated 2} (φ : V _[2]₂) : ev1₂ φ ⟶ ev2₂ φ :=
+  ⟨V.map δ0₂.op φ,
+    map_map_of_eq V (SimplexCategory.δ_comp_δ (i := 0) (j := 1) (by decide)).symm,
+    map_map_of_eq V rfl⟩
+
+/-- The arrow in the ReflQuiver `OneTruncation₂ V` of a 2-truncated simplicial set arising from the
+1st face of a 2-simplex. -/
+def ev02₂ {V : SSet.Truncated 2} (φ : V _[2]₂) : ev0₂ φ ⟶ ev2₂ φ :=
+  ⟨V.map δ1₂.op φ, map_map_of_eq V rfl, map_map_of_eq V rfl⟩
+
+/-- The arrow in the ReflQuiver `OneTruncation₂ V` of a 2-truncated simplicial set arising from the
+2nd face of a 2-simplex. -/
+def ev01₂ {V : SSet.Truncated 2} (φ : V _[2]₂) : ev0₂ φ ⟶ ev1₂ φ :=
+  ⟨V.map δ2₂.op φ, map_map_of_eq V (SimplexCategory.δ_comp_δ (j := 1) le_rfl), map_map_of_eq V rfl⟩
+
+end Truncated
+open Truncated
+
+/-- The 2-simplices in a 2-truncated simplicial set `V` generate a hom relation on the free
+category on the underlying refl quiver of `V`. -/
+inductive HoRel₂ {V : SSet.Truncated 2} :
+    (X Y : Cat.FreeRefl (OneTruncation₂ V)) → (f g : X ⟶ Y) → Prop
+  | mk (φ : V _[2]₂) :
+    HoRel₂ _ _
+      (Quot.mk _ (Quiver.Hom.toPath (ev02₂ φ)))
+      (Quot.mk _ ((Quiver.Hom.toPath (ev01₂ φ)).comp
+        (Quiver.Hom.toPath (ev12₂ φ))))
+
+theorem HoRel₂.ext_triangle {V} (X X' Y Y' Z Z' : OneTruncation₂ V)
+    (hX : X = X') (hY : Y = Y') (hZ : Z = Z')
+    (f : X ⟶ Z) (f' : X' ⟶ Z') (hf : f.edge = f'.edge)
+    (g : X ⟶ Y) (g' : X' ⟶ Y') (hg : g.edge = g'.edge)
+    (h : Y ⟶ Z) (h' : Y' ⟶ Z') (hh : h.edge = h'.edge) :
+    HoRel₂ _ _
+      ((Quotient.functor _).map (.cons .nil f))
+      ((Quotient.functor _).map (.cons (.cons .nil g) h)) ↔
+    HoRel₂ _ _
+      ((Quotient.functor _).map (.cons .nil f'))
+      ((Quotient.functor _).map (.cons (.cons .nil g') h')) := by
+  cases hX
+  cases hY
+  cases hZ
+  congr! <;> apply OneTruncation₂.Hom.ext <;> assumption
+
+/-- The type underlying the homotopy category of a 2-truncated simplicial set `V`. -/
+def hoFunctor₂Obj (V : SSet.Truncated.{u} 2) : Type u :=
+  Quotient (C := Cat.freeRefl.obj (ReflQuiv.of (OneTruncation₂ V))) (HoRel₂ (V := V))
```

Comment:
This allows the use of dot notation `V.homotopyCategory`:
```suggestion
def _root_.SSet.Truncated.homotopyCategory (V : SSet.Truncated.{u} 2) : Type u :=
  Quotient (HoRel₂ (V := V))
```
(An abbrev could be done in the `SSet` namespace so that we could also use `X.homotopyCategory` when `X : SSet`.)